ads/improvedigital.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript, checkData} from '../src/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function improvedigital (global, data) {
+  checkData(data, ['width', 'height', 'placement', 'optin', 'keyvalue']);
+     
+const customKey = (typeof data.keyvalue !== 'undefined') ? '&' + data.keyvalue : '';
+    
+    const url = 'https://ad.360yield.com' +
+      '/adj?' +
+      'p=' + data.placement +
+      '&w=' + data.width +
+      '&h=' + data.height +
+      '&optin=' + data.optin +
+       customKey +
```

Comment:
This will require special encoding since it looks like `keyvalue` is in the form `key=value`. Most likely you'd have to split it into two based on the first `=` and recombine again using `encodeURIComponent`.